tools/c7n_tencentcloud/c7n_tencentcloud/actions/core.py

```diff
@@ -0,0 +1,39 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import pytz
+from c7n.exceptions import PolicyValidationError
+
+from c7n.actions import BaseAction
+
+
+class TencentCloudBaseAction(BaseAction):
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super().__init__(data, manager, log_dir)
+        self.resource_type = self.manager.get_model()
+
+        self.endpoint = self.resource_type.endpoint
+        self.service = self.resource_type.service
+        self.version = self.resource_type.version
+        self.region = self.manager.config.region
+        self.action = ""
+
+        try:
+            self.tz = pytz.timezone(self.data.get("tz", "utc"))
```

Comment:
this probably belongs in a subclass tz is fairly specific to a few action use cases versus generically applicable.

Reply:
yes, we have removed it.